lib/screens/add_plate_weights.dart

```diff
@@ -0,0 +1,118 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:wger/providers/plate_weights.dart';
+
+class AddPlateWeights extends StatefulWidget {
+  const AddPlateWeights({super.key});
+
+  @override
+  State<AddPlateWeights> createState() => _AddPlateWeightsState();
+}
+
+class _AddPlateWeightsState extends State<AddPlateWeights> {
+
+  @override
+  Widget build(BuildContext context) {
+    return Consumer<PlateWeights> (
+      builder:(context,plateProvider,child)=> Scaffold (
+        appBar: AppBar (
+          title: const Text('Enter Custom Plate Weights'),
+        ),
+        body: Column (
+          children: [
+            Row (
+              mainAxisAlignment: MainAxisAlignment.center,
+              children: [
+                const Text('Kg or lbs'),
+                DropdownButton (
+                  onChanged: (newValue){
+                    plateProvider.clear();
+                    if((newValue!) != plateProvider.unitOfPlate) {
+                      plateProvider.unitChange();
+                    }   
+                  }, 
+                  items: ['kg','lbs'].map((unit){
+                    return DropdownMenuItem<String> (
+                      value: unit,
+                      child: Text(unit),
+                    );
+                  }).toList(),
+                ),
+              ],  
+            ),
+            SingleChildScrollView (
+              scrollDirection: Axis.horizontal,
+              child: Row (
+                children: plateProvider.unitOfPlate == 'kg'
```

Comment:
both branches are basically the same, you can just do plateProvider.unitOfPlate == 'kg' ? '$number kg' : '$number lbs' within the Text widget